modern/src/reports/ReportFilter.js

```diff
@@ -2,17 +2,27 @@ import React, { useState } from 'react';
 import {
   FormControl, InputLabel, Select, MenuItem, Button, TextField, Grid, Typography,
 } from '@material-ui/core';
-import { useSelector } from 'react-redux';
+import { useDispatch, useSelector } from 'react-redux';
 import moment from 'moment';
+import { devicesActions } from '../store';
 import t from '../common/localization';
 
-const ReportFilter = ({ children, handleSubmit, showOnly }) => {
+const ReportFilter = ({
+  children, handleSubmit, showOnly, replay,
+}) => {
   const devices = useSelector((state) => Object.values(state.devices.items));
+  const dispatch = useDispatch();
+
   const [deviceId, setDeviceId] = useState();
   const [period, setPeriod] = useState('today');
   const [from, setFrom] = useState(moment().subtract(1, 'hour'));
   const [to, setTo] = useState(moment());
 
+  const handleDeviceChange = (e) => {
+    setDeviceId(e.target.value);
+    dispatch(devicesActions.select({ id: e.target.value }));
```

Comment:
What is this used for?

Reply:
To show the name of the selected device above the slider.

Reply:
We need to find another way. You are reusing global state that's not designed for it. Also, this has to be more localized to this feature.

Reply:
Done.

Reply:
So, do we still need this code?